tests/analyze/method_linker/test_python_matcher.py

```diff
@@ -0,0 +1,123 @@
+                                                                                mock_functions, mock_call,
+                                                                                mock_var_declarations)
+
+        expected_matched_function = 'pandas_function1'
+
+        self.assertEqual(matched_function, expected_matched_function)
+
+    def test_match_call_with_other_class_functions__returns_none__when_no_match(self):
+        mock_module = Mock()
+        mock_call = 'df.method1'
+        mock_classes = {'pandas': 'class1',
+                        'DataFrame': 'class2',
+                        'numpy': 'class3'}
+        mock_functions = {'x.method1': 'x_function1',
+                          'pandas.method1': 'pandas_function1',
+                          'pandas.method2': 'pandas_function2',
+                          'pandas.method3': 'pandas_function3',
+                          'DataFrame.method1': 'df_function1',
+                          'DataFrame.method2': 'df_function2'}
+        mock_var_declarations = {'pd': 'pandas'}
+        method1_func_object = Mock()
+        method1_func_object.__qualname__ = 'pandas.method1'
+        method1_func_object.__module__ = 'module1'
+        mock_class_members = [('method1', method1_func_object), ('method2', Mock()), ('method3', Mock())]
+        python_matcher.inspect.getmembers = Mock(return_value=mock_class_members)
+
+        matched_function = python_matcher.match_call_with_other_class_functions(mock_module, mock_classes,
+                                                                                mock_functions, mock_call,
+                                                                                mock_var_declarations)
+
+        expected_matched_function = None
+
+        self.assertEqual(matched_function, expected_matched_function)
+
+    def test_match_call_with_other_class_functions__returns_function__when__module_matched(self):
+        mock_module = Mock()
+        mock_call = 'pd.date_range'
+        mock_classes = {'pandas': 'class1',
+                        'DataFrame': 'class2',
+                        'numpy': 'class3'}
+        mock_functions = {'indexes.date_range': 'indexes_function1',
+                          'pandas.method1': 'pandas_function1',
+                          'pandas.method2': 'pandas_function2',
+                          'DataFrame.method1': 'df_function1',
+                          'DataFrame.method2': 'df_function2'}
+        mock_var_declarations = {'pd': 'pandas'}
+        method1_func_object = Mock()
+        method1_func_object.__qualname__ = 'date_range'
+        method1_func_object.__module__ = 'pandas.core.indexes.datetime'
+        mock_class_members = [('date_range', method1_func_object), ('method2', Mock()), ]
+        python_matcher.inspect.getmembers = Mock(return_value=mock_class_members)
+
+        matched_function = python_matcher.match_call_with_other_class_functions(mock_module, mock_classes,
```

Comment:
in addition to testing the individual new function you added, should we test python_match_examples as that is the function that given an example, gives the source code mappings?